### ISSUE TITLE
Bug: Compiler ICEs with very simple (though incorrect) code.

### ISSUE THREAD

**Issue by crlf0710** [salsa-rs/salsa#335]
lib.rs:
```rust
#[salsa::interned]
struct XXX{
    d: i32,
}
```
Cargo.toml:
...
```rust
[dependencies]
salsa = {git = "https://github.com/salsa-rs/salsa", package = "salsa-2022"}
```

**Comment by crlf0710** [comment]
Zulip topic https://salsa.zulipchat.com/#narrow/stream/145099-general/topic/Rustc.20ICE.20with.20improper.20use.20of.20salsa.2E

**Comment by Skepfyr** [comment]
@crlf0710 Can you still repro this? I can't get anything surprising to happen.

**Comment by crlf0710** [comment]
@Skept I still can: the output is this:
```text
   Compiling dummy-salsa v0.1.0 (L:\ThirdParty\dummy-salsa)
thread 'rustc' panicked at 'index out of bounds: the len is 0 but the index is 0', compiler\rustc_errors\src\emitter.rs:1838:43
stack backtrace:
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.62.1 (e092d0b6b 2022-07-16) running on x86_64-pc-windows-msvc

note: compiler flags: --crate-type bin -C embed-bitcode=no -C debuginfo=2 -C incremental

note: some of the compiler flags provided by cargo are hidden

query stack during panic:
end of query stack
error: could not compile `dummy-salsa`
```
Frankly this seems something is wrong from the rustc side too. I just haven't had time to really dig into it yet, and it seems low priority, so i just leave it here for a little while.

**Comment by Skepfyr** [comment]
This is a bug in stable that's been fixed in nightly by https://github.com/rust-lang/rust/pull/100228. I'm tempted to close this given it's only in an error case and will be fixed in 1.65.

**Comment by crlf0710** [comment]
Good to know! Let's close this then.
